src/map/script.cpp

```diff
@@ -8782,6 +8782,136 @@ BUILDIN_FUNC(getequipuniqueid)
 	return SCRIPT_CMD_SUCCESS;
 }
 
+/*==========================================
+  * Get the item information with the specified unique id
+  * and assign it to the given variable
+  * Return true on success or false otherwise
+  * uniqueid_getiteminfo(<"item_unique_id">,<array_variable>{,<char_id>})
+  *------------------------------------------*/
+BUILDIN_FUNC(uniqueid_getiteminfo)
+{
+	TBL_PC *sd;
+
+	if (!script_charid2sd(4, sd)) {
+		script_pushint(st, false);
+		return SCRIPT_CMD_FAILURE;
+	}
+
+	unsigned long long item_uniqueid = atoll(script_getstr(st, 2));
+
+	if (!item_uniqueid) {
+		ShowError("buildin_uniqueid_delitem: unknown item (unique_id=%s).\n", script_getstr(st, 2));
+		script_pushint(st, false);
+		return SCRIPT_CMD_FAILURE;
+	}
+
+	struct script_data* data = script_getdata(st, 3);
+
+	if (!data_isreference(data)) {
+		ShowError("buildin_uniqueid_getiteminfo: not a variable\n");
+		script_reportdata(data);
+		st->state = END;
+		return SCRIPT_CMD_FAILURE;
+	}
+
+	const char *name = reference_getname(data);
+	int32 id = reference_getid(data);
+	int32 idx = reference_getindex(data);
+	struct reg_db *ref = reference_getref(data);
+
+	if (not_server_variable(*name) && !sd) {
+		ShowError("buildin_uniqueid_getiteminfo: Cannot use a player variable '%s' if no player is attached.\n", name);
+		script_reportdata(data);
+		st->state = END;
+		return SCRIPT_CMD_FAILURE;
+	}
+
+	if (is_string_variable(name)) {
+		ShowError("buildin_uniqueid_getiteminfo: Cannot Assign String variable '%s'.\n", name);
+		script_reportdata(data);
+		st->state = END;
+		return SCRIPT_CMD_FAILURE;
+	}
+
+	int i;
+
+	ARR_FIND(0, MAX_INVENTORY, i, sd->inventory.u.items_inventory[i].unique_id == item_uniqueid);
+
+	if (i >= MAX_INVENTORY) {
+		ShowError("buildin_uniqueid_getiteminfo: Item not found (unique_id=%s).\n", script_getstr(st, 2));
```

Comment:
the uniqueid string was casted to `item_uniqueid`, so just use it than use this reduntant `script_getstr(st, 2)`
```suggestion
		ShowError("buildin_uniqueid_getiteminfo: Item not found (unique_id=%llu).\n", item_uniqueid);
```